cmd/timescale-prometheus/main.go

```diff
@@ -329,6 +340,54 @@ func parseFlags() (*config, error) {
 	return cfg, nil
 }
 
+var errTimedOut = fmt.Errorf("timed out waiting to connect to DB")
+
+func waitForDb(cfg *config) error {
+	maxTimeout := 60 * time.Second
+	addr := fmt.Sprintf("%s:%d", cfg.pgmodelCfg.Host, cfg.pgmodelCfg.Port)
+
+	tryConnect := func() (done bool, err error) {
```

Comment:
why is this a closure instead of a top-level function? I'd prefer this to be a top-level func